test/unit/math/mix/core/operator_addition_test.cpp

```diff
@@ -1,303 +1,37 @@
-  y.d_.val_ = 1.0;
-
-  fvar<fvar<var> > z = x + y;
-  EXPECT_FLOAT_EQ(1, z.val_.val_.val());
-  EXPECT_FLOAT_EQ(0, z.val_.d_.val());
-  EXPECT_FLOAT_EQ(1, z.d_.val_.val());
-  EXPECT_FLOAT_EQ(0, z.d_.d_.val());
-
-  AVEC p = createAVEC(y.val_.val_);
-  VEC g;
-  z.val_.val_.grad(p,g);
-  EXPECT_FLOAT_EQ(1, g[0]);
-}
-TEST(AgradMixOperatorAddition,FvarFvarVar_FvarFvarVar_2ndDeriv_x) {
-  using stan::math::fvar;
-  using stan::math::var;
-
-  fvar<fvar<var> > x;
-  x.val_.val_ = 0.5;
-  x.val_.d_ = 1.0;
-  fvar<fvar<var> > y;
-  y.val_.val_ = 0.5;
-  y.d_.val_ = 1.0;
-
-  fvar<fvar<var> > z = x + y;
-
-  AVEC p = createAVEC(x.val_.val_,y.val_.val_);
-  VEC g;
-  z.val_.d_.grad(p,g);
-  EXPECT_FLOAT_EQ(0, g[0]);
-  EXPECT_FLOAT_EQ(0, g[1]);
-}
-TEST(AgradMixOperatorAddition,FvarFvarVar_FvarFvarVar_2ndDeriv_y) {
-  using stan::math::fvar;
-  using stan::math::var;
-
-  fvar<fvar<var> > x;
-  x.val_.val_ = 0.5;
-  x.val_.d_ = 1.0;
-  fvar<fvar<var> > y;
-  y.val_.val_ = 0.5;
-  y.d_.val_ = 1.0;
-
-  fvar<fvar<var> > z = x + y;
-
-  AVEC p = createAVEC(x.val_.val_,y.val_.val_);
-  VEC g;
-  z.d_.val_.grad(p,g);
-  EXPECT_FLOAT_EQ(0, g[0]);
-  EXPECT_FLOAT_EQ(0, g[1]);
-}
-TEST(AgradMixOperatorAddition,FvarFvarVar_Double_2ndDeriv) {
-  using stan::math::fvar;
-  using stan::math::var;
-
-  fvar<fvar<var> > x;
-  x.val_.val_ = 0.5;
-  x.val_.d_ = 1.0;
-  double y(0.5);
-
-  fvar<fvar<var> > z = x + y;
-
-  AVEC p = createAVEC(x.val_.val_);
-  VEC g;
-  z.val_.d_.grad(p,g);
-  EXPECT_FLOAT_EQ(0, g[0]);
-}
-
-TEST(AgradMixOperatorAddition,Double_FvarFvarVar_2ndDeriv) {
-  using stan::math::fvar;
-  using stan::math::var;
-
-  double x(0.5);
-  fvar<fvar<var> > y;
-  y.val_.val_ = 0.5;
-  y.d_.val_ = 1.0;
-
-  fvar<fvar<var> > z = x + y;
-
-  AVEC p = createAVEC(y.val_.val_);
-  VEC g;
-  z.d_.val_.grad(p,g);
-  EXPECT_FLOAT_EQ(0, g[0]);
-}
-TEST(AgradMixOperatorAddition,FvarFvarVar_FvarFvarVar_3rdDeriv_y) {
-  using stan::math::fvar;
-  using stan::math::var;
-
-  fvar<fvar<var> > x;
-  x.val_.val_ = 0.5;
-  x.val_.d_ = 1.0;
-  x.d_.val_ = 1.0;
-  fvar<fvar<var> > y;
-  y.val_.val_ = 0.5;
-  y.d_.val_ = 1.0;
-  y.val_.d_ = 1.0;
-
-  fvar<fvar<var> > z = x + y;
-
-  AVEC p = createAVEC(x.val_.val_,y.val_.val_);
-  VEC g;
-  z.d_.d_.grad(p,g);
-  EXPECT_FLOAT_EQ(0, g[0]);
-  EXPECT_FLOAT_EQ(0, g[1]);
-}
-TEST(AgradMixOperatorAddition,FvarFvarVar_Double_3rdDeriv) {
-  using stan::math::fvar;
-  using stan::math::var;
-
-  fvar<fvar<var> > x;
-  x.val_.val_ = 0.5;
-  x.val_.d_ = 1.0;
-  x.d_.val_ = 1.0;
-  double y(0.5);
-
-  fvar<fvar<var> > z = x + y;
+#include <stan/math.hpp>
+#include <test/unit/math/mix/mat/util/autodiff_tester.hpp>
+
+struct op_add_f {
+  template <typename T1, typename T2>
+  static typename boost::math::tools::promote_args<T1, T2>::type
+  apply(const T1& x1, const T2& x2) {
+    return x1 + x2;
+  }
+};
 
-  AVEC p = createAVEC(x.val_.val_);
-  VEC g;
-  z.d_.d_.grad(p,g);
-  EXPECT_FLOAT_EQ(0, g[0]);
-}
+TEST(foo, framework) {
```

Comment:
Want to change`foo` to something like `math`? We have these legacy names which were can shed, but maybe with something more descriptive than a placeholder.